### DISCUSSION TITLE
Invalid create() Unique constraint failed on the fields: (id), How to fix with relational attributes

### DISCUSSION THREAD

**Discussion by zorarkdouble8** [prisma/prisma#14771] [Q&A] [answered]
I understand that there's a similar discussion [here](https://github.com/prisma/prisma/discussions/5256), but I don't know how to fix this issue with relational fields. 

Basically, the **@default(autoincrement())** attribute filling the id field and creating an error because that value already exists in the database. So, I have to let the database assign the is value. In order to do this I would have to make the id optional, but I can't because of the @relational and the @id. I could remove the @id and replace it with @unique but the problem still stands.

```
model page {
  id          Int       @id @default(autoincrement())
  storyId     Int?
  fileId      Int?
  name        String?   @db.VarChar
  pagePath    String?   @db.VarChar
  endingNum   Int?
  description String?   @db.VarChar
  imgPath     String?   @db.VarChar
  useName     Boolean?
  file        file?     @relation(fields: [fileId], references: [id], onDelete: NoAction, onUpdate: NoAction)
  story       story?    @relation(fields: [storyId], references: [id], onDelete: NoAction, onUpdate: NoAction)
  display     display[]
  info        info[]

  @@index([fileId], map: "fki_page_fileId_fkey")
  @@index([storyId], map: "fki_page_storyId_fkey")
}
```

Note, the database I'm using is PostgreSQL

**Comment by austincrim** [comment, marked as answer]
Hey @zorarkdouble8 :wave:,

Have you tried [the workaround](https://github.com/prisma/prisma/discussions/5256#discussioncomment-1191352) in that discussion?

**Reply by zorarkdouble8** [reply]
Only looked at the answer, thanks!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
